### ISSUE TITLE
about the 'imageckpt/eegfeat_all_0.6875.pth'

### ISSUE THREAD

**Issue by xuxiran** [prajwalsingh/EEGStyleGAN-ADA#8]
Thank you very much for your excellent work and the open code!

After reading your paper and closed issues, I have reproduced most results in your paper except "EEGClip.py", which I will try to reproduce in a few days.

However, I have a question about "EEGStyleGAN-ADA_CVPR40".

The question is about the 'imageckpt/eegfeat_all_0.6875.pth'. I downloaded the file from your provided link and generated the good pictures. However, could I replace the 'eegfeat_all_0.6875.pth' with some other result I reproduced in "EEG2Feat"? I have many files when I reproduce the "EEG2Feat". For example, "eegfeat_all_0.9754464285714286.pth", seems better than the file you provided. So why do we need to use 'eegfeat_all_0.6875.pth' when generating pictures?

Thank you very much for your excellent work and the open code again!

**Comment by xuxiran** [comment]
Thank you very much for your kind explanation!
It is very convincing, and I will try to reproduce it today.

I found that I forgot to ask a question, in the "config.py" of "EEG2Feat_Unseen", there is a new dataset. 
For example, "
base_path       = '/path to data folder/'
train_path      = 'data/filter_eeg_imagenet40_cvpr_2017_raw/train/*/*'
validation_path = 'data/filter_eeg_imagenet40_cvpr_2017_raw/test/*/*'
test_path       = 'data/filter_eeg_imagenet40_cvpr_2017_raw/*/*/*'
"
I am wondering how to get the dataset "filter_eeg_imagenet40_cvpr_2017_raw".  
From my understanding, it seems that this dataset should be the same as the previous dataset ("EEG2Feat"). Therefore, I directly used the "eeg_imagenet40_cvpr2017_raw" dataset.
For example, "
base_path       = '/path to data folder/eeg_imagenet40_cvpr_2017_raw/'
train_path      = '/train/*'
validation_path = '/test/*'
test_path       = '/*/*'
"

Is it correct for me to do this? Or perhaps I should preprocess the raw dataset?

**Comment by prajwalsingh** [comment]
@xuxiran , yes, it's the same dataset used for EEG2Feat. Sorry for the confusion due to the code. It is not a clean code.

**Comment by xuxiran** [comment]
Hello, I came again. 
I have finished running the code about "Image2EEG". One more question.
When I ran the "train.py", the "Val KMeans score Proj" was always lower than 0.32.
![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/48015859/bfc9bd5c-d34f-4a9f-9690-360c1422a3a3)

However, when I ran the "evaluate.py", the result was 0.67, which is similar to the result shown in " eegfeat_all_0.6875.pth".

Did I make a mistake in understanding the work here? Why is there a situation where the training set is close to 1 and the validation set is only 0.32, but the results of the test set are still relatively good (0.67).

**Comment by prajwalsingh** [comment]
Hi @xuxiran , I think there might be some issue with network depth or hyper-parameters because when we train the network, we obtain the proposed accuracy over the validation set. We are attaching the snapshot for the same.

![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/22240769/b574c5ec-80d8-4b80-a81c-e824b6e4619a)

**Comment by xuxiran** [comment]
Hello, @prajwalsingh I came again.
I tried running the code on another server and this is the result.
![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/48015859/eb80bf60-e10c-4342-b504-49b75413268e)

![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/48015859/a6ac09c3-79a9-4752-98df-b330048ad0c9)
it seems lower and lower...

Maybe you are right. I think there might be some issue with network depth or hyper-parameters. However, do you mind providing some possible hyper-parameters? I can have a try. (I am using the parameters downloaded from GitHub)

**Comment by prajwalsingh** [comment]
Hi @xuxiran , I re-run the code on my system and again got almost similar accuracy, around ~59%+ (training is still ongoing). I have re-uploaded the code again although there is no change from the previous code. You can find the snapshot for the same.

I hope you are using the raw dataset only, in this work, we have focused on the raw CVPR40 dataset only. 

![image](https://github.com/prajwalsingh/EEGStyleGAN-ADA/assets/22240769/fc151f26-3d4a-4de3-9d61-eda0459922fb)
